ecc_enclave/enclave/shim.cpp

```diff
@@ -250,6 +251,15 @@ void get_public_state_by_partial_composite_key(
     unmarshal_values(values, (const char*)json, len);
 }
 
+void del_state(const char* key, shim_ctx_ptr_t ctx)
+{
+    ctx->write_set.erase(key);
+    ctx->del_set.insert(key);
+
+    ocall_del_state(key, ctx->u_shim_ctx);
```

Comment:
(this is unrelated to the PR)

Why do we do this ocall? -- we do the same for the `put_state`.

The ocall for reading is necessary to grab the data. However, the ocalls for putting and deleting keys essentially update the transaction's rwset in Fabric, which we are not using (if I recall correctly). In fact, FPC maintains its own rwset, which is validated and replayed in the second step (of the two-step transaction commit procedure).

There may still be one reason for these ocall: checking for errors, and returning any error to the enclave.
The big problem is that any feedback/error would not be trustworthy. This is likely a showstopper for such error handling.

Reply:
very good observation. We should look into that in ideally remove the ocall for `put_state` and `del_state` if really not needed. This may positively chaincode execution performance.